src/napari_imagej/widgets/parameter_widgets.py

```diff
@@ -3,14 +3,71 @@
 They should align with a SciJava ModuleItem that satisfies some set of conditions.
 """
 import importlib
+from functools import lru_cache
 from typing import Any, List
 
 from magicgui.types import ChoicesType
-from magicgui.widgets import ComboBox, Container, PushButton, request_values
+from magicgui.widgets import (
+    ComboBox,
+    Container,
+    FloatSpinBox,
+    PushButton,
+    SpinBox,
+    request_values,
+)
 from napari import current_viewer
 from napari.layers import Layer
 from napari.utils._magicgui import get_layers
 
+from napari_imagej.java import jc
+
+
+@lru_cache(maxsize=None)
+def numeric_type_widget_for(cls: type):
+    # Set sensible defaults for interfaces
+    if cls == jc.RealType or cls == jc.NumericType:
+        cls = jc.DoubleType.class_
+    elif cls == jc.IntegerType:
+        cls = jc.LongType.class_
+
+    # Return a SpinBox implementation for integer types
+    if issubclass(cls, jc.IntegerType):
+
+        class Widget(SpinBox):
```

Comment:
We could theoretically use dynamic inheritance to make only one class.

https://stackoverflow.com/questions/21060073/dynamic-inheritance-in-python

Reply:
Implemented in bc77b32